lib/CustomFields/components/CustomFieldsForm/components/CustomFieldsAccordions/CustomFieldsAccordions.js

```diff
@@ -0,0 +1,198 @@
+import {
+  injectIntl,
+  intlShape,
+} from 'react-intl';
+import {
+  AccordionSet,
+} from '@folio/stripes-components';
+
+import DraggableFieldAccordion from '../../../DraggableFieldAccordion';
+import {
+  textboxShape,
+  textareaShape,
+  checkboxShape,
+  permissionsShape,
+} from '../../../../shapes';
+import { fieldTypes } from '../../../../constants';
+
+import styles from './CustomFieldsAccordions.css';
+
+const propTypes = {
+  accordions: PropTypes.object.isRequired,
+  changeFieldValue: PropTypes.func.isRequired,
+  fields: PropTypes.shape({
+    length: PropTypes.number.isRequired,
+    map: PropTypes.func.isRequired,
+    remove: PropTypes.func.isRequired,
+    value: PropTypes.arrayOf(PropTypes.shape({
+      id: PropTypes.string.isRequired,
+      values: PropTypes.oneOfType([
+        textboxShape,
+        textareaShape,
+        checkboxShape,
+      ]).isRequired,
+    })).isRequired,
+  }).isRequired,
+  getFormState: PropTypes.func.isRequired,
+  intl: intlShape,
+  onDeleteClick: PropTypes.func.isRequired,
+  onDragEnd: PropTypes.func.isRequired,
+  onToggleAccordion: PropTypes.func.isRequired,
+  permissions: permissionsShape.isRequired,
+};
+
+const CustomFieldsAccordions = ({
+  fields,
+  permissions,
+  accordions,
+  onToggleAccordion,
+  onDragEnd,
+  onDeleteClick,
+  changeFieldValue,
+  getFormState,
+  intl,
+}) => {
+  const getCustomFieldNameById = (id) => {
+    const customField = fields.value.find(cf => cf.id === id);
+    if (!customField) {
+      return '';
+    }
+
+    return customField.values.name;
+  };
+
+  const onDragStart = (start, provided) => {
+    provided.announce(intl.formatMessage({
+      id: 'stripes-smart-components.customFields.dragDrop.lifted',
+    }, {
+      name: getCustomFieldNameById(start.draggableId),
+      position: start.source.index + 1,
+    }));
+  };
+
+  const onDragUpdate = (update, provided) => {
+    const message = update.destination
+      ? intl.formatMessage({
+        id: 'stripes-smart-components.customFields.dragDrop.moving',
+      }, {
+        name: getCustomFieldNameById(update.draggableId),
+        position: update.destination.index + 1,
+      })
+      : intl.formatMessage({
+        id: 'stripes-smart-components.customFields.dragDrop.notOverDroppable',
+      });
+
+    provided.announce(message);
+  };
+
+  const handleDragEnd = (result, provided) => {
+    const name = getCustomFieldNameById(result.draggableId);
+
+    const message = result.destination
+      ? intl.formatMessage({
+        id: 'stripes-smart-components.customFields.dragDrop.moved',
+      }, {
+        name,
+        start: result.source.index + 1,
+        end: result.destination.index + 1,
+      })
+      : intl.formatMessage({
+        id: 'stripes-smart-components.customFields.dragDrop.returned',
+      }, {
+        name,
+        position: result.source.index + 1,
+      });
+
+    provided.announce(message);
+
+    onDragEnd(result, provided);
+  };
+
+  return (

```

Comment:
@BogdanDenis a lot of things are happening in this render. Would it be possible for you to refactor it and split it up a bit?